auth/token-managers/jwt-token-manager.ts

```diff
@@ -175,36 +202,27 @@ export class JwtTokenManager {
       throw new Error(err);
     }
 
-    this.expireTime = this.calculateTimeForNewToken(accessToken);
-    this.tokenInfo = extend({}, tokenResponse);
-  }
-
-  /**
-   * Decode the access token and calculate the time to request a new token.
-   *
-   * A time buffer prevents the edge case of the token expiring before the request could be made.
-   * The buffer will be a fraction of the total time to live - we are using 80%
-   *
-   * @param accessToken - JSON Web Token received from the service
-   * @private
-   * @returns {void}
-   */
-  private calculateTimeForNewToken(accessToken): number {
     // the time of expiration is found by decoding the JWT access token
     // exp is the time of expire and iat is the time of token retrieval
-    let timeForNewToken;
     const decodedResponse = jwt.decode(accessToken);
-    if (decodedResponse) {
-      const { exp, iat } = decodedResponse;
-      const fractionOfTtl = 0.8;
-      const timeToLive = exp - iat;
-      timeForNewToken = exp - (timeToLive * (1.0 - fractionOfTtl));
-    } else {
+    if (!decodedResponse) {
       const err = 'Access token recieved is not a valid JWT'
       logger.error(err);
       throw new Error(err);
     }
 
-    return timeForNewToken;
+    const { exp, iat } = decodedResponse;
+    // There are no required claims in JWT
+    if (!exp || !iat) {
```

Comment:
This is just extra bullet-proofing for the case where the JWT does not have an `exp` or `iat` claim.